centreon/www/include/configuration/configGenerate/xml/moveFiles.php

```diff
@@ -348,15 +348,21 @@
                     }
                 }
             } else {
-                passthru("echo 'SENDCFGFILE:" . $host['id'] . "' >> $centcore_pipe", $return);
+                passthru(
+                    escapeshellcmd("echo 'SENDCFGFILE:{$host['id']}'") . ' >> ' . escapeshellcmd($centcore_pipe),
+                    $return
+                );
                 if ($return) {
                     throw new Exception(_("Could not write into centcore.cmd. Please check file permissions."));
                 }
                 if (!isset($msg_restart[$host["id"]])) {
                     $msg_restart[$host["id"]] = "";
                 }
                 if (count($listBrokerFile) > 0) {
-                    passthru("echo 'SENDCBCFG:" . $host['id'] . "' >> $centcore_pipe", $return);
+                    passthru(
+                        escapeshellcmd("echo 'SENDCBCFG:{$host['id']}") . ' >> ' . escapeshellcmd($centcore_pipe),
```

Comment:
You can use `escapeshellarg` when you have only one argument.
```suggestion
                        escapeshellcmd("echo 'SENDCBCFG:{$host['id']}") . ' >> ' . escapeshellarg($centcore_pipe),
```